Kitodo/src/test/java/org/kitodo/production/services/data/ProcessServiceIT.java

```diff
@@ -100,6 +100,28 @@ public void shouldCountAllDatabaseRowsForProcesses() throws Exception {
         assertEquals("Processes were not counted correctly!", Long.valueOf(3), amount);
     }
 
+    @Test
+    public void shouldSaveProcess() throws Exception {
+        Process parent = new Process();
+        parent.setTitle("Parent");
+
+        Process process = new Process();
+        process.setTitle("Child");
+        process.setParent(parent);
+        parent.getChildren().add(process);
+
+        processService.save(process);
+
+        Process foundProcess = processService.getById(process.getId());
+
+        assertEquals("Child process was not found in database!", "Child", foundProcess.getTitle());
+        assertEquals("Parent process was not assigned to child!", "Parent", process.getParent().getTitle());
+
+        Integer parentId = foundProcess.getParent().getId();
+
+        processService.remove(parentId);
```

Comment:
Why are you only removing the parent?

Reply:
Parent is the owner, it removes own children.

Reply:
Oh well.. i'm not sure, if this is the behaviour we want :(
I'm going to check this

Reply:
I also don't know, but it is default behavior for Hibernate. When parent is removed, child are automatically gone also.